File: 1007-minimum-domino-rotations-for-equal-row/1007-minimum-domino-rotations-for-equal-row.java
class Solution {
    public int minDominoRotations(int[] tops, int[] bottoms) {
        HashMap<Integer, Integer> mMap = new HashMap<>();
        for (int i = 0; i < tops.length; i++) {
            if (mMap.containsKey(tops[i])) {
                mMap.put(tops[i], mMap.get(tops[i]) + 1);
            } else {
                mMap.put(tops[i], 1);
            }
            if (mMap.containsKey(bottoms[i])) {
                mMap.put(bottoms[i], mMap.get(bottoms[i]) + 1);
            } else {
                mMap.put(bottoms[i], 1);
            }
        }
                
        int number = 0;
        for (Integer val: mMap.keySet()) {
            if (mMap.get(val) >= tops.length) {
                number = val;
                break;
            }
        }
        System.out.println(number);
        if (number == 0) {
            return -1;
        }
        
        int countOne = 0;
        int countTwo = 0;
        for (int i = 0; i < tops.length; i++) {
            if (tops[i] != number && bottoms[i] == number) {
                countOne++;
            } else if (tops[i] != number) {
                return -1;
            }
            if (bottoms[i] != number && tops[i] == number) {
                countTwo++;
            } else if (bottoms[i] != number) {
                return -1;
            }
        }
        return Math.min(countOne, countTwo);
        
        
                
    }
}

File: 1007-minimum-domino-rotations-for-equal-row/1007-minimum-domino-rotations-for-equal-row.py
class Solution(object):
    def minDominoRotations(self, tops, bottoms):
        """
        :type tops: List[int]
        :type bottoms: List[int]
        :rtype: int
        """
        arr = [0 for i in range(7)]
        for i in tops:
            arr[i] += 1
        for i in bottoms:
            arr[i] += 1
            
        length = len(tops)
        targetNum = -1
        for i in range(len(arr)):
            if arr[i] >= length:
                targetNum = i
                break
        if targetNum == -1:
            return -1
        
        resultTop = 0
        resultBottom = 0
        for i in range(len(tops)):
            if tops[i] != targetNum:
                if bottoms[i] == targetNum:
                    resultTop += 1
                else:
                    return -1
            if bottoms[i] != targetNum:
                if tops[i] == targetNum:
                    resultBottom += 1
                else:
                    return -1
        return min(resultTop, resultBottom)
                    
        
        
        